simulate/simulate.cc

```diff
@@ -2597,6 +2605,16 @@ void Simulate::Render() {
     ShowFigure(this, viewport, &figure);
```

Comment:
These calls to `ShowFigure`, `ShowOverlayText` and `ShowImage` are being made from Simulate's `Render` function. `Render` is called from the `RenderLoop` after unlocking Simulate's lock (this helps with parallelism and makes the viewer run smoother).

However, `ShowFigure`, `ShowOverlayText` and `ShowImage` use vectors, `user_figures_`, `user_text_`, and `user_images_` which can be modified concurrently by the user code. The result is that undefined behavior can occur.

See top level review comment about how we can resolve this issue in general.